coalib/core/Core.py

```diff
@@ -145,6 +149,91 @@ def finish_task(bear,
                     exc_info=ex)
 
 
+def initialize_dependencies(bears):
+    """
+    Initializes and returns a ``DependencyTracker`` instance together with a
+    set of bears ready for scheduling.
+
+    This function acquires, processes and registers bear dependencies
+    accordingly using a consumer-based system, where each dependency bear has
+    only a single instance per section and file-dictionary.
+
+    The bears set returned accounts for bears that have dependencies and
+    excludes them accordingly. Dependency bears that have themselves no further
+    dependencies are included so the dependency chain can be processed
+    correctly.
+
+    :param bears:
+        The set of bears to run that serve as an entry-point.
+    :return:
+        A tuple with ``(dependency_tracker, bears_to_schedule)``.
+    """
+    # Pre-collect bears in a set as we use them more than once. Especially
+    # remove duplicate instances.
+    bears = set(bears)
+
+    dependency_tracker = DependencyTracker()
+
+    # For a consumer-based system, we have a situation which can be visualized
+    # with a graph:
+    #
+    # (section1, file_dict1) (section1, file_dict2) (section2, file_dict2)
+    #       |       |                  |                      |
+    #       V       V                  V                      V
+    #     bear1   bear2              bear3                  bear4
+    #       |       |                  |                      |
+    #       V       V                  |                      |
+    #  BearType1  BearType2            -----------------------|
+    #       |       |                                         |
+    #       |       |                                         V
+    #       ---------------------------------------------> BearType3
+    #
+    # We need to traverse this graph and instantiate dependency bears
+    # accordingly, one per section.
+
+    # Group bears by sections and file-dictionaries. These will serve as
```

Comment:
A section does not contain the actual file-dict, it just contains the `files`-setting which contains a (unevaluated) glob. Actually some section settings will be treated specially in our current design (like `bears`, `files`, `ignore`, `default_actions` etc), as some of them are bound to the CLI, and some do control how bears shall be run outside of the scope of the bear itself.
This will get more separated with aspects, though this is still a way to go ;)